lib/internal/http2/core.js

```diff
@@ -3302,12 +3288,16 @@ function connectionListener(socket) {
 
   socket[kServer] = this;
 
+  // NGHTTP2_INITIAL_CONNECTION_WINDOW_SIZE default is 64KB, but we want to allow more data to be
+  // buffered without applying backpressure.
+  session.setLocalWindowSize(512 * 1024);
+
   this.emit('session', session);
 }
 
 function initializeOptions(options) {
   assertIsObject(options, 'options');
-  options = { ...options };
+  options = { ...getDefaultSettings(), ...options };
```

Comment:
We should actually apply the default settings and not just assume that they are the same as nghttp2 will do.